src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -6401,7 +6412,7 @@ private IOperation VisitNoneOperationExpression(IOperation operation)
             // We turn the interpolated string into a call to create the handler type, a series of append calls (potentially with branches, depending on the
             // handler semantics), and then evaluate to the handler flow capture temp.
 
-            SpillEvalStack();
+            SpillEvalStack(spillDeclarationExpressions: true);
```

Comment:
>spillDeclarationExpressions: true [](http://example.com/codeflow?start=27&length=33)

This doesn't feel sufficient. This parameter will make the difference only for entries that haven't been spilled already. And there is no guarantee that that will be the case. For example, if an argument following a ```DeclarationExpression``` has a control flow, it will case spilling for the ```DeclarationExpression``` and after that this call won't do anything about it. It should be easy to come up with a test case that exposes this issue.
Some options to address this that come to mind:
- Unconditionally unwrap ```DeclarationExpression```s during spilling. If we can run into tuple-like ```DeclarationExpression```s, perhaps we can skip those.
- Preemptively unwrap ```DeclarationExpression```s when we push arguments to the stack.
- Handle spilling of ```DeclarationExpression```out of order on demand once we realize that we need to use it multiple times due to the interpolation.
We can discuss our options in more details offline.